src/pages/proposal-details/index.tsx

```diff
@@ -0,0 +1,328 @@
+import { observer } from 'mobx-react-lite';
+import React, { FunctionComponent, useCallback, useMemo } from 'react';
+import { CanvasJSChart } from 'canvasjs-react-charts';
+import DataTable from 'src/components/insync/data-table';
+import { useStore } from 'src/stores';
+import { useParams } from 'react-router-dom';
+import { Loader } from 'src/components/common/loader';
+import CircularProgress from 'src/components/insync/circular-progress';
+import { VictoryContainer, VictoryPie, VictoryTooltip } from 'victory';
+import { voteCalculation } from 'src/utils/vote-calculation';
+import { statusColor } from 'src/utils/color-formatter';
+import { useValidators } from 'src/hooks/use-validators';
+import ValidatorName from 'src/components/common/validator-name';
+import { CellRendererProps, ColumnDef } from 'src/components/insync/data-table/types';
+import moment from 'moment';
+
+interface ProposalDetailParam {
+	proposalId: string;
+}
+
+interface ParsedValidator {
+	moniker: string;
+	identity: number;
+	status: number;
+	proposalId: string;
+	address: string;
+}
+
+const URL_REGEX = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)/;
+const EMPTY_TABLE_DATA: any[][] = [];
+
+const ValidatorCell: FunctionComponent<CellRendererProps<any, ParsedValidator>> = ({ data, rowIndex }) => (
+	<ValidatorName index={rowIndex} moniker={data.moniker} identity={data.identity} />
+);
+
+const AnswerCell: FunctionComponent<CellRendererProps<any, ParsedValidator>> = observer(({ data, rowIndex }) => {
+	console.log('========\n', 'data', data, '\n========');
+	const { chainStore, queriesStore } = useStore();
+
+	const votingData = queriesStore
+		.get(chainStore.current.chainId)
+		.rebus.queryValidatorVoting.getVotingRecord(data.proposalId, data.address);
+
+	console.log('========\n', 'votingData', votingData.response, '\n========');
+	return <div>data.status</div>;
+});
+
+const ValidatorTable: FunctionComponent<ProposalDetailParam> = ({ proposalId }) => {
+	const { inProgress, validatorList } = useValidators();
+
+	const validatorData = useMemo(() => {
+		const activeValidators = validatorList?.filter((item: any) => item.status === 3);
+
+		console.log('========\n', 'activeValidators', activeValidators, '\n========');
+
+		return activeValidators && activeValidators.length
+			? activeValidators.map(item => {
+					const parsedItem: ParsedValidator = {
+						moniker: item.description.moniker,
+						identity: item.description.identity,
+						status: item.status,
+						address: item.operator_address,
+						proposalId,
+					};
+
+					return parsedItem;
+			  })
+			: EMPTY_TABLE_DATA;
+	}, [proposalId, validatorList]);
+
+	const columnDefs: ColumnDef<ParsedValidator>[] = [
```

Comment:
This can be defined outside this component too since its static

Reply:
removed